sdk/python/feast/constants.py

```diff
@@ -21,135 +21,194 @@ class AuthProvider(Enum):
+
+    #: Enable user authentication to Feast Core
+    ENABLE_AUTH: str = "False"
+
+    #: Auth token for user authentication to Feast
+    AUTH_TOKEN: str = ""
+
+    #: Path to certificate(s) to secure connection to Feast Core
+    CORE_SERVER_SSL_CERT: str = ""
+
+    #: Default Feast Serving URL
+    SERVING_URL: str = "localhost:6566"
+
+    #: Enable or disable TLS/SSL to Feast Serving
+    SERVING_ENABLE_SSL: str = "False"
+
+    #: Path to certificate(s) to secure connection to Feast Serving
+    SERVING_SERVER_SSL_CERT: str = ""
+
+    #: Default Feast Job Service URL
+    JOB_SERVICE_URL: str = ""
+
+    #: Enable or disable TLS/SSL to Feast Job Service
+    JOB_SERVICE_ENABLE_SSL: str = "False"
+
+    #: Path to certificate(s) to secure connection to Feast Job Service
+    JOB_SERVICE_SERVER_SSL_CERT: str = ""
+
+    #: Enable or disable control loop for Feast Job Service
+    JOB_SERVICE_ENABLE_CONTROL_LOOP: str = "False"
+
+    #: Default connection timeout to Feast Serving, Feast Core, and Feast Job Service (in seconds)
+    GRPC_CONNECTION_TIMEOUT: str = "10"
+
+    #: Default gRPC connection timeout when sending an ApplyFeatureSet command to Feast Core (in seconds)
+    GRPC_CONNECTION_TIMEOUT_APPLY: str = "600"
+
+    #: Default timeout when running batch ingestion
+    BATCH_INGESTION_PRODUCTION_TIMEOUT: str = "120"
+
+    #: Time to wait for historical feature requests before timing out.
+    BATCH_FEATURE_REQUEST_WAIT_TIME_SECONDS: str = "600"
+
+    #: Authentication Provider - Google OpenID/OAuth
+    AUTH_PROVIDER: str = "google"
+
+    #: Spark Job launcher
+    SPARK_LAUNCHER: str = "dataproc"  # standalone, dataproc, emr
+
+    #: Feast Spark Job ingestion jobs staging location
+    SPARK_STAGING_LOCATION: str = ""
+
+    #: Feast Spark Job ingestion jar file
+    SPARK_INGESTION_JAR: str = "https://storage.googleapis.com/feast-jobs/spark/ingestion/feast-ingestion-spark-develop.jar"
+
+    #: Spark resource manager master url
+    SPARK_STANDALONE_MASTER: str = "local[*]"
+
+    #: Directory where Spark is installed
+    SPARK_HOME: str = ""
+
+    #: Dataproc cluster to run Feast Spark Jobs in
+    DATAPROC_CLUSTER_NAME: str = ""
+
+    #: Project of Dataproc cluster
+    DATAPROC_PROJECT: str = ""
+
+    #: Region of Dataproc cluster
+    DATAPROC_REGION: str = ""
+
+    #: File format of historical retrieval features
+    HISTORICAL_FEATURE_OUTPUT_FORMAT: str = "parquet"
+
+    #: File location of historical retrieval features
+    HISTORICAL_FEATURE_OUTPUT_LOCATION: str = ""
+
+    #: Default Redis host
+    REDIS_HOST: str = "localhost"
+
+    #: Default Redis port
+    REDIS_PORT: str = "6379"
+
+    #: Enable or disable TLS/SSL to Redis
+    REDIS_SSL: str = "False"
+
+    #: Enable or disable StatsD
+    STATSD_ENABLED: str = "False"
+
+    #: Default StatsD port
+    STATSD_HOST: str = ""
+
+    #: Default StatsD port
+    STATSD_PORT: str = ""
+
+    #: IngestionJob DeadLetter Destination
```

Comment:
What are the possible options?

Reply:
Updated.

Reply:
Again `gs` supported by dataproc launcher, `s3` by emr.
We do not include s3 support in dataproc launcher and vice versa